stream-chat-android-offline/src/test/java/io/getstream/chat/android/offline/repository/facade/RepositoryFacadeIntegrationTests.kt

```diff
@@ -0,0 +1,42 @@
+package io.getstream.chat.android.offline.repository.facade
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth
+import io.getstream.chat.android.offline.integration.BaseRepositoryFacadeIntegrationTest
+import io.getstream.chat.android.offline.randomMessage
+import io.getstream.chat.android.test.randomString
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+internal class RepositoryFacadeIntegrationTests : BaseRepositoryFacadeIntegrationTest() {
+
+    @Test
+    fun `Given a message in the database When persisting the updated message Should store the update`() = runBlocking {
+        val id = randomString()
+        val originalMessage = randomMessage(id = id)
+        val updatedText = randomString()
+        val updatedMessage = originalMessage.copy(text = updatedText)
+
+        repositoryFacade.insertMessages(listOf(originalMessage), cache = false)
+        repositoryFacade.insertMessages(listOf(updatedMessage), cache = false)
+        val result = repositoryFacade.selectMessage(id)
+
+        Truth.assertThat(result).isNotNull()
+        Truth.assertThat(result!!.text).isEqualTo(updatedText)
+    }
+
+    @Test
+    fun `Given a message When persisting the message Should store html field`() = runBlocking {
+        val html = randomString()
+        val id = randomString()
+        val message = randomMessage(id = id, html = html)
+
+        repositoryFacade.insertMessages(listOf(message), cache = false)
+        val result = repositoryFacade.selectMessage(id)
+
+        Truth.assertThat(result).isNotNull()
+        Truth.assertThat(result!!.html).isEqualTo(html)
```

Comment:
What about to assert all fields of our message? :thinking: 
Because our `Message` class is a dataclass, check if the entity has the same values should be as simple as:
```
Truth.assertThat(result).isEqualTo(expectedMessage)
```

It will help us to verify any new field we add to our `Message` entity

Reply:
Not all the fields are stored in our database

Reply:
Overall, added 3 missing fields which were not persisted: `html`, `showInChannel` and `silent`. The `online` filed is ignored on purpose, there is no point to persist fields with quickly expiring values. 

`totalUnreadCount` and `unreadChannels` are also not persisted as they will be overwritten by null values coming from the server.